src/entity/SquadPostModeration.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  ManyToOne,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+import { Post } from './posts';
+import { Source } from './Source';
+import { User } from './user';
+
+export enum SquadPostModerationStatus {
+  Approved = 'approved',
+  Rejected = 'rejected',
+  Pending = 'pending',
+}
+
+@Entity()
+export class SquadPostModeration {
+  @PrimaryColumn({ type: 'text' })
+  id: string;
+
+  @Column({ type: 'text' })
+  sourceId: string;
+
+  @ManyToOne('Source', (source: Source) => source.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  source: Promise<Source>;
+
+  @Column({ type: 'text' })
+  status: SquadPostModerationStatus;
+
+  @Column({ type: 'text', nullable: true })
+  createdById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'SET NULL',
+  })
+  createdBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatedById: string | null;
+
+  @ManyToOne('User', (user: User) => user.id, {
+    lazy: true,
+    onDelete: 'SET NULL',
+  })
+  moderatedBy: Promise<User>;
+
+  @Column({ type: 'text', nullable: true })
+  moderatorMessage: string | null;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Column({ type: 'text', nullable: true })
+  postId: string | null;
+
+  @ManyToOne('Post', (post: Post) => post.id, {
+    lazy: true,
+    onDelete: 'CASCADE',
+  })
+  post: Promise<Post>;
+
+  @Column({ type: 'text' })
+  type: string;
+
+  @Column({ type: 'text', nullable: true })
+  title?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  titleHtml: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  content?: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  contentHtml: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  image: string | null;
+
+  @Column({ type: 'text', nullable: true })
+  sharedPostId?: string | null;
+
+  @ManyToOne('Post', (post: Post) => post.id, {
+    lazy: true,
+    onDelete: 'SET NULL',
+  })
+  sharedPost: Promise<Post>;
```

Comment:
Why not just cascade on delete? (same for other above)

Reply:
True, this one should probably cascade.

@capJavert Which do you mean "other above" ? The one for user?

If that's the case, I think that even if the user gets deleted, the post may be interesting for the squad and they might want to keep it, I talked with Lee about potentially reassigning those to Ghost user.

What do you think ?

Reply:
yeah but here you are describing the relation of what happens wtih SquadPostModeration when post gets deleted, which should always delete moderation items when post is deleted?

Reply:
I don't think user would want anything of his posted if you delete an account as well, so maybe better to just keep it simple and avoid this case if it was not mentioned in the RFC or product spec.